core/vesting/vesting.go

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) 2023 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE.VEGA file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package vesting
+
+import (
+	"context"
+
+	"code.vegaprotocol.io/vega/core/assets"
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/logging"
+	proto "code.vegaprotocol.io/vega/protos/vega"
+	"golang.org/x/exp/maps"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -destination mocks/mocks.go -package mocks code.vegaprotocol.io/vega/core/vesting Collateral,ActivityStreakVestingMultiplier,Broker,Assets
+
+type Collateral interface {
+	TransferVestedRewards(
+		ctx context.Context, transfers []*types.Transfer,
+	) ([]*types.LedgerMovement, error)
+}
+
+type ActivityStreakVestingMultiplier interface {
+	Get(party string) num.Decimal
+}
+
+type Broker interface {
+	Send(events events.Event)
+}
+
+type Assets interface {
+	Get(assetID string) (*assets.Asset, error)
+}
+
+type PartyRewards struct {
+	// the amounts per assets still being locked in the
+	// account and not available to be released
+	// this is a map of:
+	// asset -> (remainingEpochLock -> Amount)
+	Locked map[string]map[uint64]*num.Uint
+	// the current part of the vesting account
+	// per asset available for vesting
+	Vesting map[string]*num.Uint
+}
+
+type Engine struct {
+	log *logging.Logger
+
+	c      Collateral
+	asvm   ActivityStreakVestingMultiplier
+	broker Broker
+	assets Assets
+
+	minTransfer num.Decimal
+	baseRate    num.Decimal
+
+	state map[string]*PartyRewards
+}
+
+func New(
+	log *logging.Logger,
+	c Collateral,
+	asvm ActivityStreakVestingMultiplier,
+	broker Broker,
+	assets Assets,
+) *Engine {
+	log = log.Named(namedLogger)
+
+	return &Engine{
+		log:         log,
+		c:           c,
+		asvm:        asvm,
+		broker:      broker,
+		assets:      assets,
+		baseRate:    num.MustDecimalFromString("0.01"),
```

Comment:
why do you need this initialised here? it should never be used with this value anyways

Reply:
removed.